domain/transfer.go

```diff
@@ -87,13 +88,21 @@ func (TransferProcessHandler) HandleEvent(
 		r.FromAccountID = x.FromAccountID
 		r.ToAccountID = x.ToAccountID
 
-		s.ExecuteCommand(commands.DebitAccount{
-			TransactionID:   x.TransactionID,
-			AccountID:       x.FromAccountID,
-			TransactionType: messages.Transfer,
-			Amount:          x.Amount,
-			ScheduledDate:   x.ScheduledDate,
-		})
+		scheduled, err := time.Parse("2006-01-02", x.ScheduledDate)
+		if err != nil {
+			// TODO: Panic with some error about invalid argument
+			return err
+		}
+
+		s.ScheduleTimeout(
+			TransferAtScheduledTimeout{
+				TransactionID: x.TransactionID,
+				FromAccountID: x.FromAccountID,
+				Amount:        x.Amount,
+				ScheduledDate: x.ScheduledDate,
+			},
+			scheduled,
```

Comment:
Another option for the date conversion would be to move it to this line... something like 
```go
s.ScheduleTimeout(
    TransferAtScheduledTimeout{... ...},
    DateToTime(x.ScheduledDate),
)
```

Reply:
I basically did exactly that, using the existing helper function.